### ISSUE TITLE
Is there a way to continue training the pre-trained model?

### ISSUE THREAD

**Issue by Alvarixx** [georgw777/chesscog#26]
I have read the code and saw that in the file:

chesscog/core/training/train.py

There's a function (train_model) that can load the model from the /runs folder for training. Could I continue training the pre-trained model this way? So it can better recognize more than one chess set.

In my case, I have several, and creating a sizable dataset for each one isn't a problem for me.

Or maybe it's better to make an initial training only for my chess sets and keep training with more images + JSON this way.

I think your project works better than the newer one (https://github.com/ThanosM97/end-to-end-chess-recognition), the difference remains on the dataset.

**Comment by georg-wolflein** [comment]
Hi @Alvarixx, yes this is possible; see the following section from the README:

> Relevant documentation is available in Appendix C.2 of my [master thesis](https://github.com/georg-wolflein/chesscog-report/raw/master/report.pdf). In particular, Appendix C.2.6 contains instructions for fine-tuning the model to your own chess set, using only two input images. Be sure to run python -m chesscog.transfer_learning.create_dataset before training (a step I forgot to mention in those instructions, see https://github.com/georg-wolflein/chesscog/issues/15.

**Comment by Alvarixx** [comment]
Hi! Yes, I tested it, with one new chess set fine-tune works amazing! But if I load two different chess set fails.

The question is, if I train the model with (supposing, not possible of course) 5.000 images with his correspondent .json using photos from different sets, instead the blend.

Do you think the model will be able to work with good accuracy?

Maybe it is obvious but I'm new with CNN, even with python :)

I saw this project -> https://github.com/ThanosM97/end-to-end-chess-recognition) and I think your method is better, but they didn't tested something like this (using the same dataset), just fine-tuned.

**Comment by georg-wolflein** [comment]
Ah, I understand now. Yes, this is a great idea, and I originally planned on doing this. However, I didn’t have the time and patience to create this dataset of different real chessboards to do the finetuning on them.

If you create your own dataset with different chess boards including the JSON files it should be possible to finetune chesscog to be more general (ie work for different chessboards with the same CNN).

Good luck! If you want to share your finetuned model for others to use, I will be more than happy to link to your repository:)

**Comment by Alvarixx** [comment]
Of course yes, I will try it in the following months and share if works nice.

You did a great job!

**Comment by Alvarixx** [comment]
I've sent you two more general models.

It's trained with a set of 1,500 positions generated and labeled by me and 500 positions from your dataset. I used augmentation.

I've noticed some things that could be improved while learning and seeing how everything works:

1. The difference between white and black pieces is a matter of luminance, not color.

I solved this by training the model and sending it through recognition.py with GrayScale. It has very few errors detecting white or black pieces at the moment.

2. With the same dataset, it could be trained better.

I think the optimal solution is to create a model with 6 classes (for each piece) and a model to distinguish between white and black pieces. This way, you have twice as many images for each piece.

3. Corner detection fails too much.

No solution at the moment.

The most important thing is that I trained them with GrayScale so the models focus on certain aspects. You need to include this line in recognition.py

img = cv2.cvtColor(img, cv2.COLOR_BGR2GRAY)
img = cv2.cvtColor(img, cv2.COLOR_BGR2RGB)
